Common/src/Common.Infrastructure/IServiceCollectionExtension.cs

```diff
@@ -1,20 +1,14 @@
 ﻿using Enmeshed.Common.Infrastructure.Persistence.Repository;
-using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Enmeshed.Common.Infrastructure.Persistence.Context;
 
 namespace Enmeshed.Common.Infrastructure;
 public static class IServiceCollectionExtension
 {
-    public static IServiceCollection AddMetricStatusesRepository(this IServiceCollection services, IConfiguration configuration)
+    public static IServiceCollection AddMetricStatusesRepository(this IServiceCollection services, Action<MetricStatusesDapperContext> configureOptions)
```

Comment:
You're using the MetricStatusesDapperContext as Options. But you should introduce `MetricStatusesRepositoryOptions`, which have a property `ConnectionString`, are configured by the action and then injected into the `MetricStatusesDapperContext`